src/LuaTransformer.ts

```diff
@@ -4995,57 +5009,48 @@ export class LuaTransformer {
                         expression
                     );
                 }
-                if (!isWithinExpressionStatement) {
+                if (expression.parent.kind !== ts.SyntaxKind.ExpressionStatement) {
                     throw TSTLErrors.ForbiddenLuaTableSetExpression(expression);
                 }
                 break;
         }
     }
 
-    protected transformLuaTableExpressionStatement(
-        node: ts.ExpressionStatement & { expression: ts.CallExpression } & {
-            expression: { expression: ts.PropertyAccessExpression };
-        }
-    ): tstl.VariableDeclarationStatement | tstl.AssignmentStatement {
-        const methodName = node.expression.expression.name.text;
-        const signature = this.checker.getResolvedSignature(node.expression);
-        const tableName = (node.expression.expression.expression as ts.Identifier).text;
-        const luaTable = tstl.createIdentifier(tableName);
-        const params = this.transformArguments((node.expression as ts.CallExpression).arguments, signature);
+    protected transformLuaTableExpressionAsExpressionStatement(expression: ts.CallExpression): tstl.Statement {
+        const methodName = this.getLuaTablePropertyName(expression.expression);
+        const signature = this.checker.getResolvedSignature(expression);
+        const tableAccessExpression = this.transformExpression(expression.expression);
```

Comment:
`AccessExpression` can be quite misleading since it can be an identifier as well